tests/fixtures/helpers.ts

```diff
@@ -0,0 +1,257 @@
+        success = true;
+        socket.end();
+      }
+
+      if (message.error) {
+        socket.end();
+      }
+    };
+
+    const buffers: Buffer[] = [];
+    let remainingBytes = 0;
+
+    socket.on('data', (data) => {
+      while (true) {
+        if (remainingBytes === 0) {
+          const index = data.indexOf(':');
+
+          buffers.push(data);
+
+          if (index === -1) {
+            return;
+          }
+
+          const buffer = Buffer.concat(buffers);
+          const bufferIndex = buffer.indexOf(':');
+
+          buffers.length = 0;
+          remainingBytes = Number(buffer.subarray(0, bufferIndex).toString());
+
+          if (!Number.isFinite(remainingBytes)) {
+            throw new Error('Invalid state');
+          }
+
+          data = buffer.subarray(bufferIndex + 1);
+        }
+
+        if (data.length < remainingBytes) {
+          remainingBytes -= data.length;
+          buffers.push(data);
+          break;
+        } else {
+          buffers.push(data.subarray(0, remainingBytes));
+
+          const buffer = Buffer.concat(buffers);
+          buffers.length = 0;
+
+          const json = JSON.parse(buffer.toString());
+          queueMicrotask(() => {
+            onMessage(json);
+          });
+
+          const remainder = data.subarray(remainingBytes);
+          remainingBytes = 0;
+
+          if (remainder.length === 0) {
+            break;
+          } else {
+            data = remainder;
+          }
+        }
+      }
+    });
+  });
+};
+
+export async function loadContext(browserName: string) {
+  const pathToExtension = getPathToExtension(browserName);
+  let context: BrowserContext | undefined;
+  if (browserName === 'chromium') {
+    context = await chromium.launchPersistentContext('', {
+      headless: false, // headless isn't well supported with extensions
+      args: [
+        `--disable-extensions-except=${pathToExtension}`,
+        `--load-extension=${pathToExtension}`,
+      ],
+    });
+  } else if (browserName === 'firefox') {
+    const RDP_PORT = 12345;
+    context = await firefox.launchPersistentContext('', {
+      headless: false,
+      args: ['-start-debugger-server', String(RDP_PORT)],
+      firefoxUserPrefs: {
+        'devtools.debugger.remote-enabled': true,
+        'devtools.debugger.prompt-connection': false,
+        'extensions.webextensions.uuids': JSON.stringify({
+          // See `browser_specific_settings.gecko.id` in manifest.json
+          'tech@interledger.org': FIREFOX_ADDON_UUID,
+        }),
+      },
+    });
+    await loadFirefoxAddon(RDP_PORT, 'localhost', pathToExtension);
+  }
+
+  if (!context) {
+    throw new Error('Unknown browser: ' + browserName);
+  }
+
+  // Note that loading this directly via config -> use({ storageState }) doesn't
+  // work correctly with our browser context. So, we addCookies manually.
+  const { cookies } = await readFile(authFile, 'utf8').then(JSON.parse);
+  await context.addCookies(cookies);
```

Comment:
This is how we added the previously used cookie - after logging into Rafiki Money?

Reply:
Yes.